test/clj_momo/lib/map_test.clj

```diff
@@ -22,4 +22,7 @@
               :ab "ok"
               :ac {:aca "ok"
                    :acb "ok"
-                   :acc "ok"}}})))
+                   :acc "ok"}}}))
+  (is (= {:a :b} (sut/deep-merge nil {:a :b})))
+  (is (= {:a :b} (sut/deep-merge {:a :b} nil)))
+  (is (nil? (sut/deep-merge nil nil))))
```

Comment:
I would try to at least give the values returned for

```clojure
(deep/merge {:a {:b :c}} nil {:a nil} {:a {:b {:x y}}})
```

And certainly more examples for the 4 possible cases:

- the value in m1 is a hash-map but in m2 it is:
  - non-existent, 
  - nil, 
  - not a hash-map
  - a hash-map
- the value in m1 is not a hash-map but in m2 it is:
  - non-existent
  - nil
  - a hash-map
  - not a hash-map